### PR TITLE
Feature/tm10-13

### PR DESCRIPTION
**Pull request by Sefanovskis-Artjoms**

### CODE REVIEW

File: app/card-info/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+"use client";
+import { useParams } from "next/navigation";
+import { useTodo } from "../../../context/TodoContext";
+import TodoCard from "../../components/TodoCard";
+
+export default function CardDetailsPage() {
+  const { cardData } = useTodo();
+  const params = useParams();
+  const { id } = params;
+  const idNumber = +id;
```
Comment (wanpii) [inline app/card-info/[id]/page.tsx]: what is the reason for line 13?

File: app/components/Navigation.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          <Link
+            href="/"
+            className="text-gray-800 font-semibold bg-gray-300 hover:bg-gray-400 hover:text-gray-900 py-2 px-4 rounded-md transition-colors duration-300"
+          >
+            To do list
+          </Link>
+        </li>
+        <li>
+          <Link
+            href="/AddCard"
```
Comment (wanpii) [inline app/components/Navigation.tsx]: don't use different types of urls for example here you have CamelCase url, but next file it's /card-info/${information.id}
the standard is kebab case, so I suggest sticking to it, this applies to everything else, try to be consistent with your code.

File: app/components/AddTodoCard.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      status: "In progress",
+      type: "Feature",
+      title: "",
+      description: "",
+    },
+  });
+
+  const onFormSubmit = (data: TodoFormInputsStructure) => {
+    const newItem: TodoCardInfoStructure = {
+      id: Date.now(),
```
Comment (wanpii) [inline app/components/AddTodoCard.tsx]: Don't use this kind of solution, instead you can use a library or generate a random ID yourself

File: types.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+export interface TodoCardInfoStructure {
```
Comment (wanpii) [inline types.ts]: you can remove Structure word from naming, TodoCardInfo and TodoFormInput is enough

**Review by wanpii — CHANGES_REQUESTED** [review]

File: app/card-info/[id]/page.tsx (hunk 1 above)
Comment (Sefanovskis-Artjoms) [reply app/card-info/[id]/page.tsx]: It is for comparison, to get card info on the line 14. ID from URL is string that must be converted to number before comparison can happen.

File: context/TodoContext.tsx [hunk 5]
```diff
@@ -2,19 +2,19 @@
...
 
-const TodoContext = createContext<TodoContextStructure | undefined>(undefined);
+const TodoContext = createContext<TodoCardContext | undefined>(undefined);
 
 export const TodoProvider = ({ children }: { children: React.ReactNode }) => {
-  const [cardData, setCardData] = useState<TodoCardInfoStructure[]>([
+  const [cardData, setCardData] = useState<TodoCardInfo[]>([
     {
-      id: 1,
+      id: "1",
```
Comment (wanpii) [inline context/TodoContext.tsx:17]: now I see why you used const idNumber = +id 😆

**Review by wanpii — APPROVED** [review]

**Pull request merged**